pkg/eval/sys/sys.go

```diff
@@ -0,0 +1,25 @@
+// Package sys exposes variables and functions that deal with the specific
+// system being run on. Such as the OS name and CPU architecture.
+package sys
+
+import (
+	"runtime"
+
+	"github.com/elves/elvish/pkg/eval"
+	"github.com/elves/elvish/pkg/eval/vars"
+)
+
+//elvdoc:var arch
+//
+// The architecture of the platform; e.g., amd64, arm, ppc. This is read-only.
+
+//elvdoc:var platform
+//
+// The name of the platform; e.g., darwin (macOS), linux. This is read-only.
+
+var Ns = eval.NewNs()
```

Comment:
The following is not succinct:

```go
var Ns = eval.Ns{
  "arch": ...,
  ...
}
```

Reply:
I assume you meant "more succinct". Fair enough. Done. Although, as a grey beard this is a situation where I think the use of `init()` is preferable so that when functions are added in the future only the `eval.NewNS()` is changed to include an invocation of the `AddGoFns()` method. I hope you're not suggesting the chaining two or more mutating methods in this situation is good style. If the overall sequence was very short I could live with it; and, in fact, have written such code.   The problem is that style tends to devolve into a "bike shedding" discussion which is never good.